packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -31,6 +35,106 @@ type BaseButtonWithIconProps = BaseButtonCommonProps & {
 
 export type BaseButtonProps = BaseButtonWithIconProps | BaseButtonWithoutIconProps;
 
+const StyledBaseText = styled(BaseText)(
```

Comment:
Need to do this since we don't have an enhancer component for `Spacing` which we need to add before the button. I've added support for extending BaseText with `styled` (this means accepting a `style` prop for it to work on native). I could have done the same with `Icon` component but that would means accepting and passing down `style` prop for _every_ new icon component that is created (Eg. CreditCardIcon, ChevronDownIcon, etc.).

Reply:
Not for the consumers. We're using `BaseText` inside `BaseButton` which needs to pass down "extra styles" (padding) for `BaseText` when there is an icon.